aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -30,6 +31,21 @@ struct AT_CUDA_API CUDAEvent {
   CUDAEvent(unsigned int flags = DEFAULT_FLAGS)
   : flags_{flags} { }
 
+  CUDAEvent(const cudaIpcEventHandle_t* handle)
+  : CUDAEvent(getCurrentCUDAStream(), handle) { }
+
+  CUDAEvent(const CUDAStream& stream, const cudaIpcEventHandle_t* handle) {
```

Comment:
I think it might be better to just take a `at::Device` or `at::DeviceIndex` since you don't need a stream, just a device.